apps/chat/src/components/Common/ApplicationDialog.tsx

```diff
@@ -36,688 +33,561 @@ import Modal from '@/src/components/Common/Modal';
-  const files = useAppSelector(FilesSelectors.selectFiles);
-  const featuresDataInputRef = useRef<HTMLTextAreaElement>(null);
-  const [features, setFeatures] = useState('');
-  const [featuresDataError, setFeaturesDataError] = useState('');
-  const [maxAttachments, setMaxAttachments] = useState(0);
-  const [completionUrl, setCompletionUrl] = useState('');
-  const [completionUrlError, setCompletionUrlError] = useState('');
-  const [submitted, setSubmitted] = useState(false);
-  const [nameError, setNameError] = useState<string | null>(null);
-  const [versionError, setVersionError] = useState<string | null>(null);
-  const [iconError] = useState<string | null>(null);
-  const nameInputRef = useRef<HTMLInputElement>(null);
-  const descriptionInputRef = useRef<HTMLTextAreaElement>(null);
-  const [filterParams, setFilterParams] = useState<string[]>([]);
-  const [isDeleteModalOpen, setIsDeleteModalOpen] = useState(false);
-  const [isPublishing, setIsPublishing] = useState(false);
-
-  const entity = selectedApplication && {
-    name: selectedApplication.name,
-    id: ApiUtils.decodeApiUrl(
-      selectedApplication.id ||
-        (selectedApplication as unknown as { application: string }).application,
-    ),
-    folderId: getFolderIdFromEntityId(selectedApplication.name),
-  };
-
-  const resetForm = () => {
-    setName('');
-    setVersion('');
-    setDescription('');
-    setFeatures('');
-    setMaxAttachments(0);
-    setCompletionUrl('');
-    setLocalLogoFile(undefined);
-    setDeleteLogo(false);
-    setFilterParams([]);
-  };
+}) => {
+  const {
+    register,
+    handleSubmit,
+    setValue,
+    clearErrors,
+    trigger,
+    control,
+    formState: { errors, isValid },
+  } = useForm<FormData>({
+    mode: 'onChange',
+    reValidateMode: 'onChange',
+  });
 
-  const loading = useAppSelector((state) => state.application.loading);
+  const { t } = useTranslation(Translation.Chat);
 
-  const handleChangeFilterParams = useCallback((filterParams: string[]) => {
-    setFilterParams(filterParams);
-  }, []);
+  const dispatch = useAppDispatch();
 
-  const handleClose = useCallback(() => {
-    setSubmitted(false);
-    resetForm();
-    onClose(false);
-  }, [onClose]);
+  const selectedApplication = useAppSelector(
+    ApplicationSelectors.selectApplicationDetail,
+  );
+  const files = useAppSelector(FilesSelectors.selectFiles);
 
-  const nameOnBlurHandler = (e: FocusEvent<HTMLInputElement>) => {
-    const newName = prepareEntityName(e.target.value.trim(), {
-      forRenaming: true,
-    });
-    const pattern = /^[^!@#$^*()+]{2,160}$/;
-
-    if (!pattern.test(newName)) {
-      setNameError(
-        t(
-          'Name should be 2 to 160 characters long and should not contain special characters',
-        ),
-      );
-    } else {
-      setNameError(null);
-    }
-    setName(newName);
-    onBlur(e);
-  };
+  const selectedApplicationId = getSelectedApplicationId(selectedApplication);
```

Comment:
You can add into `CustomApplicationModel` property `application?: string;` and use
```suggestion
  const selectedApplicationId = selectedApplication?.application ?? selectedApplication?.id;
```
istead of `getSelectedApplicationId`

Reply:
I think this approach is better. `application` field can't be with `id` simultaneously. In this case we can precisely use interfaces for custom and public apps

Reply:
Why do we have different type `PublicCustomApplicationModel` with different property `application` for public application? `PublicCustomApplicationModel` isn't used anywhere directly and `application` can be optional field of `CustomApplicationModel`

Reply:
Because it's invalid. This object can't have `id` and `application` fields at once. Typescript will start give hints on fields that may not exist. If we will make `application` field optional, we need to make `id` field optional to, because if we have `application` we don't have id and if we have`id` we don't have `application`.